### DISCUSSION TITLE
How to decode the txs?

### DISCUSSION THREAD

**Discussion by 0xZonh** [cosmos/community#37] [Announcements]
I call the Rest API ：[http://xxx.xxx.xxx.xxx:26657/block?height=1940777],  it return the block meta data and with follow txs. But if I want to retrive the transaction detail infomation, I need call the Rest API [http://xxx.xxx.xxx.xxx:26657/txs?decoce="xxxxx"] . How to local decode method the txs data?
`"data": {
        "txs": [          "2AEoKBapCkGoo2GaChTUQ7UQqXgc06K5wEDT+Gyr05c4ihIU0fSoEE1eNXfAwiqlecpdhQl7yQIaDwoFdWF0b20SBjEwNDI1NRIQCgoKBXVhdG9tEgExEOj7AxpqCibrWumHIQLpPzaUxCoE1Y/kEy1/p7WV8rxy6benYj8uB+dQAwD4WhJA4HMKaAJ5boV8st1XAXewTm0mcqo/awE1cNljnICCvE0KTTWpaVQvdTyDv/faU+LhUy5OQxb1mWBWQhNEv0PyaCIRdzR4bTluLDQ4LDgsNTAwMDA=",
         ]
 }`

**Comment by facundomedica** [comment]
You need to get the TxConfig for the app you are talking to and use the TXDecoder:

```
txb, err := s.clientCtx.TxConfig.TxDecoder()(req.TxBytes)
```

**Comment by puneet2019** [comment]
you are calling rpc endpoints, 
if you are willing to call rest endpoints: there is GET `/cosmos/tx/v1beta1/txs/block/{height}` for getting block txs decoded :  from cosmos-0.45.x
or if you want to just decode the data try: POST `/cosmos/tx/v1beta1/decode` [example from persistence testnet2](https://hopp.sh/r/FQPDSJcY57b4) : from cosmos-0.47.x

there should be grpc methods too: [protos here](https://github.com/cosmos/cosmos-sdk/blob/main/proto/cosmos/tx/v1beta1/service.proto#L37-L52)
